GetPMWallet/handler.ts

```diff
@@ -44,37 +78,143 @@ type IHttpHandler = (
   userAndCitizenId: RequestCitizenToAdUserAndFiscalCode
 ) => Promise<IResponseSuccessJson<Wallet> | ErrorTypes>;
 
+const InternalServerErrorResponse = ResponseErrorInternal(
+  "Internal server error"
+);
+
+export const toApiPMWalletList = (
+  _: WalletV2ListResponse
+): Either<t.Errors, Wallet> => {
+  return Wallet.decode({
+    data: fromNullable(_.data)
+      .map(walletList =>
+        walletList.reduce((acc, wallet) => {
+          const info = wallet.info;
+          if (
+            wallet.walletType === WalletTypeEnum.Card &&
+            WalletCardInfoInput.is(info)
```

Comment:
Is this double check needed? If not we can manage each branch with a `switch`

Reply:
Yes, this check is needed because the typings of info is `object`.